ext/calendar/dow.c

```diff
@@ -35,6 +35,10 @@ int DayOfWeek(
 {
 	int dow;
 
+	if ((sdn + 1) > INT_MAX) {
```

Comment:
zend_long can be the same as int; in which case this check is removed by the compiler because it'll be UB.
Note that you can just change the formula and the if-checks a bit because we're working in mod 7 to avoid the overflow; and avoid needing to check the overflow in the first place.